code/modules/antagonists/abductor/equipment/glands/heal.dm

```diff
@@ -0,0 +1,177 @@
+	uses = -1
+	human_only = TRUE
+	icon_state = "health"
+	mind_control_uses = 3
+	mind_control_duration = 3000
+
+/obj/item/organ/heart/gland/heal/activate()
+	if(!MOB_ORGANIC in owner.mob_biotypes)
+		return
+
+	for(var/organ in internal_organs)
+		if(istype(organ, /obj/item/organ/cyberimp))
+			reject_implant(organ)
+			return
+
+	var/obj/item/organ/liver/liver = owner.getorganslot(ORGAN_SLOT_LIVER)
+	if((!liver/* && !HAS_TRAIT(owner, TRAIT_NOMETABOLISM)*/) || (liver && ((liver.damage > (liver.maxHealth / 2)) || (istype(liver, /obj/item/organ/liver/cybernetic)))))
+		replace_liver(liver)
+		return
+
+	var/obj/item/organ/lungs/lungs = owner.getorganslot(ORGAN_SLOT_LUNGS)
+	if((!lungs && !HAS_TRAIT(owner, TRAIT_NOBREATH)) || (lungs && (istype(lungs, /obj/item/organ/lungs/cybernetic))))
+		replace_lungs(lungs)
+		return
+
+	var/obj/item/organ/eyes/eyes = owner.getorganslot(ORGAN_SLOT_EYES)
+	if(!eyes || (eyes && ((HAS_TRAIT_FROM(owner, TRAIT_NEARSIGHT, EYE_DAMAGE)) || (HAS_TRAIT_FROM(owner, TRAIT_BLIND, EYE_DAMAGE)) || (istype(eyes, /obj/item/organ/eyes/robotic)))))
+		replace_eyes(eyes)
+		return
+
+	var/obj/item/bodypart/limb
+	var/list/limb_list = list(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	for(var/zone in limb_list)
+		limb = owner.get_bodypart(zone)
+		if(!limb)
+			replace_limb(zone)
+			return
+		if((limb.get_damage() >= (limb.max_damage / 2)) || (limb.status == BODYPART_ROBOTIC))
+			replace_limb(zone, limb)
+			return
+
+	if(owner.blood_volume < BLOOD_VOLUME_OKAY)
+		replace_blood()
+		return
+	if(owner.getToxLoss() > 40)
+		replace_blood()
+		return
+	var/tox_amount = 0
+	for(var/datum/reagent/toxin/T in owner.reagents.reagent_list)
+		tox_amount += owner.reagents.get_reagent_amount(T.type)
+	if(tox_amount > 10)
+		replace_blood()
+		return
+
+	var/obj/item/bodypart/chest/chest = owner.get_bodypart(BODY_ZONE_CHEST)
+	if((chest.get_damage() >= (chest.max_damage / 4)) || (chest.status == BODYPART_ROBOTIC))
+		replace_chest(chest)
+		return
+
+/obj/item/organ/heart/gland/heal/proc/reject_implant(obj/item/organ/cyberimp/implant)
+	owner.visible_message("<span class='warning'>[owner] vomits up his [implant.name]!</span>", "<span class='userdanger'>You suddenly vomit up your [implant.name]!</span>")
+	owner.vomit(0, TRUE, TRUE, 1, FALSE, FALSE, FALSE, TRUE)
+	implant.Remove(owner)
+	implant.forceMove(owner.drop_location())
+
+/obj/item/organ/heart/gland/heal/proc/replace_liver(obj/item/organ/liver/liver)
+	if(liver)
+		owner.visible_message("<span class='warning'>[owner] vomits up his [liver.name]!</span>", "<span class='userdanger'>You suddenly vomit up your [liver.name]!</span>")
+		owner.vomit(0, TRUE, TRUE, 1, FALSE, FALSE, FALSE, TRUE)
+		liver.Remove(owner)
+		liver.forceMove(owner.drop_location())
+	else
+		to_chat(owner, "<span class='warning'>You feel a weird rumble in your bowels...</span>")
+	
+	var/liver_type = /obj/item/organ/liver
+	if(owner.dna.species && owner.dna.species.mutantliver)
```

Comment:
I'm not 100% sure how abductor implants trigger, but this might runtime if implanted into a monkey.

Reply:
the gland has human_only = TRUE, meaning it doesnt trigger on carbons

Reply:
```suggestion
	if(owner?.dna?.species?.mutantliver)
```